plugin/trino-hive/src/main/java/io/trino/plugin/hive/parquet/ParquetPageSourceFactory.java

```diff
@@ -303,14 +303,15 @@ public static ReaderPageSource createPageSource(
 
     public static ParquetDataSource createDataSource(
             TrinoInputFile inputFile,
+            long length,
             OptionalLong estimatedFileSize,
             ParquetReaderOptions options,
             AggregatedMemoryContext memoryContext,
             FileFormatDataSourceStats stats)
             throws IOException
     {
         if (estimatedFileSize.isEmpty() || estimatedFileSize.getAsLong() > options.getSmallFileThreshold().toBytes()) {
-            return new TrinoParquetDataSource(inputFile, options, stats);
+            return new TrinoParquetDataSource(inputFile, length, options, stats);
```

Comment:
`length` here is supposed to be split length rather than file length and split length can be smaller than file length
I think the fix here was to just update TracingInputFile https://github.com/trinodb/trino/pull/21073